bytestring/common/test/ByteStringHexTest.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+@file:OptIn(ExperimentalStdlibApi::class)
+
+package kotlinx.io.bytestring
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class ByteStringHexTest {
```

Comment:
Please add a few test cases covering negative scenarios (where a call should fail with an exception), as you did for b64.